examples/template-ts/src/index.ts

```diff
@@ -0,0 +1,67 @@
+import { assign, createMachine, interpret, StateFrom } from 'xstate';
+import './styles.css';
+
+const eventOutputEl = document.getElementById('output-event')!;
+const valueOutputEl = document.getElementById('output-value')!;
+const contextOutputEl = document.getElementById('output-context')!;
+
+const timeMachine = createMachine<{ time?: string }>(
+  {
+    id: 'time',
+    initial: 'idle',
+    context: {
+      time: undefined
+    },
+    states: {
+      idle: {
+        on: {
+          fetch: 'pending'
+        }
+      },
```

Comment:
I really like this example. One thing for clarity might be worth capitalizing states and actions or using some type of differentiation to say that those are user-defined and not special xstate keys. especially when things like fetch already exist in javascript.